Amplify/Categories/DataStore/Model/Collection/List+LazyLoad.swift

```diff
@@ -32,20 +32,12 @@ extension List {
 
         // if the collection has no associated field, return the current elements
         guard let associatedId = self.associatedId,
-              let associatedField = self.associatedField else {
+              let associatedField = associatedField else {
             completion(.success(elements))
             return
         }
 
-        // TODO: this is currently done by specific plugin implementations (API or DataStore)
-        // How to add this name resolution to Amplify?
-        let modelName = Element.modelName
-        var name = modelName.camelCased() + associatedField.name.pascalCased() + "Id"
-        if case let .belongsTo(_, targetName) = associatedField.association {
-            name = targetName ?? name
-        }
-
-        let predicate: QueryPredicate = field(name) == associatedId
+        let predicate: QueryPredicate = field(associatedField.name) == associatedId
```

Comment:
great! lazy load will always pass the name of the associated field, and let it be resolved to the column name internally, just like how the customer would